### DISCUSSION TITLE
kubernetes compatibility chart

### DISCUSSION THREAD

**Discussion by andrezelak-wind** [docker-library/rabbitmq#683] [General]
Hello,

Are there some known compatibility issue or limitation between RabbitMQ and Kubernetes versions?
I couldn't find any warning in the RabbitMQ container release notes.
I also looked in the CI logs, they don't show the Kubernetes version used.

Thanks!

**Comment by lukebakken** [comment]
Hello - this repository is not the best place to ask this question.

> Are there some known compatibility issue or limitation between RabbitMQ and Kubernetes versions?

My guess is "no", as long as you are using a reasonably recent version of k8s. You absolutely should be using the [RabbitMQ Cluster Kubernetes Operator](https://github.com/rabbitmq/cluster-operator?tab=readme-ov-file#supported-versions), which does require k8s 1.19 or later.

I suggest using one of the [community support forums](https://www.rabbitmq.com/#support):
* https://groups.google.com/forum/#!forum/rabbitmq-users
* https://github.com/rabbitmq/cluster-operator/discussions

**Reply by andrezelak-wind** [reply]
Hello Luke,

I'll follow your advice and forward the question to those groups.
We aren't using RabbitMQ Cluster Kubernetes Operator, so I'll read about it.

Thank you for guiding me!

**Reply by michaelklishin** [reply]
The image is entirely unaware if it's deployed to Kubernetes or not.

Every RabbitMQ Streams user on Kubernetes must be [aware of this doc section](https://rabbitmq.com/memory-use.html#page-cache) and consider running on 1.25+ with cgroups v2 enabled.

The K8S RabbitMQ cluster Operator supports a wide range of K8S versions for historical reasons. It will likely begin requiring something much more modern, maybe even 1.25.0, starting next year.

**Reply by andrezelak-wind** [reply]
Hello Michael!

I believe it is a reasonable request to use Kubernetes 1.25 or newer.

Thank you!

**Reply by michaelklishin** [reply]
Note that you must also use a [modern cgroups v2-enabled distribution](https://rabbitmq.com/memory-use.html#page-cache) but with this image, you should be all set in that regard: the Ubuntu variant [uses 22.04](https://github.com/docker-library/rabbitmq/blob/master/3.12/ubuntu/Dockerfile#L10), which is a cgroups v2-enabled distribution according to our community's findings.

**Reply by andrezelak-wind** [reply]
Ok, I'll double check the cgroups v2 is enabled in the distro.
